### ISSUE TITLE
size_in_tb still expecting 2tb minimum

### ISSUE THREAD

**Issue by bernard-nguyen** [hashicorp/terraform-provider-azurerm#26301]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

### Terraform Version

1.6.6

### AzureRM Provider Version

3.107.0

### Affected Resource(s)/Data Source(s)

azurerm_netapp_pool

### Terraform Configuration Files

```hcl
resource "azurerm_netapp_pool" "example" {
  name                = "example-netapppool"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name
  account_name        = azurerm_netapp_account.example.name
  service_level       = "Premium"
  size_in_tb          = 1
}
```

### Debug Output/Panic Output

```shell
Planning failed. Terraform encountered an error while generating this plan.

Error: expected size_in_tb to be in the range (2 - 500), got 1
```

### Expected Behaviour

Pool with a size of 1TB should be created as this "feature" is in now in GA.

https://azure.microsoft.com/fr-fr/updates/general-availability-azure-netapp-files-support-for-1-tib-capacity-pools/

### Actual Behaviour

Error: expected size_in_tb to be in the range (2 - 500), got 1

### Steps to Reproduce

1.) Create TF Code
2.) Run TF apply

### Important Factoids

No

**Comment by Dergons** [comment]
We also ran into this today when trying to set up a NetApp account

**Comment by neil-yechenwei** [comment]
Thanks for raising this issue. [Azure Doc](https://learn.microsoft.com/en-us/azure/azure-netapp-files/azure-netapp-files-set-up-capacity-pool) mentioned that this new feature would be supported in new API version 2024-01-01 but it seems that it hasn't been released to [Swagger](https://github.com/Azure/azure-rest-api-specs/tree/main/specification/netapp/resource-manager/Microsoft.NetApp/stable).

**Comment by QBY-ChristianHartmann** [comment]
Thanks for merging the fix that the validation accepts 1tb now. Nonetheless the azurerm resource stillt uses the [2023-05-01](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/netapp/client/client.go) API ant therefore the bug persists!

**Comment by ldgithub007** [comment]
Yes, Plan is passed but apply is still failing even though i am using Azurerm version is 4.5.0. How to change the azurerm to use latest API version 2024-01-01 as mentioned in this Microsoft document?  
Capacity Pool Name: "pool1"): performing PoolsCreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidValueReceivedFor: An invalid value '1099511627776' was received for Pool.Size. Must be 2 tebibytes in bytes or higher

**Comment by ashishchawla-tfc** [comment]
Hi team, do we have any update on the same ?

**Comment by RickShahid** [comment]
Hi, any update on resolving this issue?

The latest Microsoft.NetApp provider API version is 2024-07-01
Here is the Azure CLI command that I use to check for released provider API versions
 
az provider show --namespace Microsoft.NetApp --query "resourceTypes[?resourceType=='netAppAccounts/capacityPools'].apiVersions[]"

**Comment by poxy91** [comment]
I got the same issue when using azurerm provider 4.8.0:
performing PoolsCreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidValueReceivedFor: An invalid value '1099511627776' was received for Pool.Size. Must be 2 tebibytes in bytes or higher

However updating to the azurerm provider to 4.24.0 solved the issue

**Comment by poxy91** [comment]
I got the same issue when using azurerm provider 4.8.0:
performing PoolsCreateOrUpdate: unexpected status 400 (400 Bad Request) with error: InvalidValueReceivedFor: An invalid value '1099511627776' was received for Pool.Size. Must be 2 tebibytes in bytes or higher

However updating to the azurerm provider to 4.24.0 solved the issue

**Comment by neil-yechenwei** [comment]
The API version is upgraded to 2025-01-01. So this issue has gone.

**Comment by mybayern1974** [comment]
Closing this issue given the feature has been supported https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/netapp_pool#size_in_tb-1
